src/Guardrails-Common/GR-Common.psm1

```diff
@@ -436,34 +456,41 @@ function Check-GAAuthenticationMethods {
     
     }
 
-    # GA UPN list has less than 2 UPN
+    # Condition: GA UPN list has less than 2 UPN
     if ($globalAdminUPNs.Count -lt 2) {
         $commentsArray += $msgTable.globalAdminMinAccnts
     }
-    # GA UPN list has > 2 UPNs and all are MFA enabled
+    # Condition: GA UPN list has > 2 UPNs and all are MFA enabled
     elseif($globalAdminUPNs.Count -ge 2 -and $mfaCounter -eq $globalAdminUPNs.Count) {
         $commentsArray += $msgTable.globalAdminMFAPassAndMin2Accnts
         $IsCompliant = $true
     }
-    # GA UPN list has > 2 UPNs and not all UPNs are MFA enabled
+    # Condition: GA UPN list has > 2 UPNs and not all UPNs are MFA enabled
     else{
-        # only one UPN is not MFA enable
-        if ( $GAUPNsMFA.Count -eq 1 ) {
-            $commentsArray += $msgTable.globalAdminAccntsMFADisabled1 -f $GAUPNsMFA.MFAComments
-        }
-        # None are MFA enabled
-        elseif ( $GAUPNsMFA.Count -eq $globalAdminUPNs.Count) {
-            $commentsArray += $msgTable.globalAdminAccntsMFADisabled3
+        # This will be used for debugging
+        if($GAUPNsMFA.Count -lt 1){
+            Write-Host "Something is wrong as GAUPNsMFA Count equals 0. This output should only execute if there is an error populating GAUPNsMFA"
         }
-        # 2 or more UPNs in the list are not MFA enabled
-        else {
-            $hiddenUPNsString = ""
-            for ($i =0; $i -lt $GAUPNsMFA.Count; $i++) {
-                $hiddenUPNsString += $GAUPNsMFA[$i].MFAComments + ", "
+        else{
+            # only one UPN is not MFA enable
+            if ( $GAUPNsMFA.Count -eq 1 ) {
+                $commentsArray += $msgTable.globalAdminAccntsMFADisabled1 -f $GAUPNsMFA.MFAComments
```

Comment:
Should this be `$GAUPNsMFA[0].MFAComments` because `GAUPNsMFA` is an array

Reply:
No, you use .Count on an array to get the size

Reply:
This works as expected as it is a single object array.
![image](https://github.com/ssc-spc-ccoe-cei/azure-guardrails-solution-accelerator/assets/147828743/ea0529d9-56f5-4e4d-83f8-ce2e3597570a)

But following the code norm I'll change it.